lab_4_summarization_textrank/main.py

```diff
@@ -155,27 +215,32 @@ class SimilarityMatrix:
     """
 
     _matrix: list[list[float]]
+    _vertices: list[Sentence]
 
     def __init__(self) -> None:
         """
         Constructs necessary attributes
         """
-        pass
+        self._matrix = []
+        self._vertices = []
 
     def get_vertices(self) -> tuple[Sentence, ...]:
         """
         Returns a sequence of all vertices present in the graph
         :return: a sequence of vertices
         """
-        pass
+        return tuple(self._vertices)
 
     def calculate_inout_score(self, vertex: Sentence) -> int:
         """
         Retrieves a number of vertices that are similar (i.e. have similarity score > 0) to the input one
         :param vertex
         :return:
         """
-        pass
+        if vertex not in self._vertices:
```

Comment:
how do you eliminate loops?

Reply:
they are being eliminated in add_edge method, in 254-255 lines